src/lib/listeners/ProgressEventListener.ts

```diff
@@ -144,12 +153,17 @@ export class EngineRunProgressSpinner extends ProgressSpinner implements Progres
 		this.isListening = false;
 	}
 
-	public listen(codeAnalyzer: CodeAnalyzer): void {
+	public listen(...codeAnalyzers: CodeAnalyzer[]): void {
+		if (codeAnalyzers.length > 1) {
+			// It's technically possible to implement multiple-core support for this spinner, but not currently necessary.
```

Comment:
If we _really_ want to, we can convert `progressMap` into an array of maps and aggregate each engine's completion percentage across all of the maps. But there's absolutely no reason for us to do that _now_, so I just decided not to.